### PR TITLE
SC-107 Set due date 

### PR DESCRIPTION
**Pull request by edwin-jebaraj**
## Changes in this PR
1. app/controllers/support/cases/evaluation_due_dates_controller.rb - Added new controller to update evaluation due date
2. app/models/support/case.rb - Added evaluation due date validation
3. app/views/support/cases/evaluation_due_dates/edit.html.erb - Added new view file to update evaluation due date
4. config/locales/en.yml - Added new language variable
5. config/locales/validation/en.yml - Added nee validation language variable
6. config/routes.rb - Added new route for evaluation due date screen
7. db/migrate/20241126122758_add_evaluation_due_date_to_support_cases.rb - Added new migration file to add new field in support_cases table
8. 9.spec/features/support/agent_can_add_evaluation_due_date_spec.rb- Added new validation spec file for evaluaton due date

### CODE REVIEW

File: app/controllers/support/cases/evaluation_due_dates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+module Support
+  class Cases::EvaluationDueDatesController < Cases::ApplicationController
+    before_action :set_current_case
+    before_action { @back_url = "/support/cases/#{params[:case_id]}#tasklist" }
```
Comment (threedaymonk) [inline app/controllers/support/cases/evaluation_due_dates_controller.rb:4]: This should use a URL helper with the [`anchor` parameter for the `#tasklist`](https://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html#:~:text=link_to%20can%20also,strings:)

```ruby
support_case_path(@current_case, anchor: "tasklist")
```

File: app/controllers/support/cases/evaluation_due_dates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def edit
+      @evaluation_due_date = @current_case
+    end
+
+    def update
+      @evaluation_due_date = Support::Case.find(params[:case_id])
+      @evaluation_due_date.assign_attributes(evaluation_due_date_params)
+
+      if @evaluation_due_date.valid?(:update_due_date_form)
+        if @evaluation_due_date.save(context: :update_due_date_form)
```
Comment (threedaymonk) [inline app/controllers/support/cases/evaluation_due_dates_controller.rb:14]: You've already loaded the case and assigned it to `@current_case` in `set_current_case`.

File: app/controllers/support/cases/evaluation_due_dates_controller.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    end
+
+    def date_param(form_param, date_field)
+      date = params.fetch(form_param, {}).permit("#{date_field}(1i)", "#{date_field}(2i)", "#{date_field}(3i)")
+      begin
+        Date.new(date["#{date_field}(1i)"].to_i, date["#{date_field}(2i)"].to_i, date["#{date_field}(3i)"].to_i)
+      rescue StandardError
+        nil
+      end
+    end
```
Comment (threedaymonk) [inline app/controllers/support/cases/evaluation_due_dates_controller.rb:43]: You shouldn't have to do all this explicitly: `@current_case.update(params.require(:support_case).permit(:evaluation_due_date))` ought to work in Rails.

**Review by threedaymonk — CHANGES_REQUESTED** [review]

File: app/controllers/support/cases/evaluation_due_dates_controller.rb (hunk 1 above)
Comment (edwin-jebaraj) [reply app/controllers/support/cases/evaluation_due_dates_controller.rb]: Updated

File: app/controllers/support/cases/evaluation_due_dates_controller.rb (hunk 2 above)
Comment (edwin-jebaraj) [reply app/controllers/support/cases/evaluation_due_dates_controller.rb]: updated

File: app/controllers/support/cases/evaluation_due_dates_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    include HasDateParams
+    def edit
+      @case_evaluation_due_date_form = CaseEvaluationDueDateForm.new(**current_case.to_h)
+    end
+
+    def update
+      @case_evaluation_due_date_form = CaseEvaluationDueDateForm.from_validation(validation)
+
+      if validation.success?
+        @current_case.update!(@case_evaluation_due_date_form.to_h)
```
Comment (threedaymonk) [inline app/controllers/support/cases/evaluation_due_dates_controller.rb:15]: It's still the same!

**Review by threedaymonk — CHANGES_REQUESTED** [review]

**Review by threedaymonk — APPROVED** [review]

**Pull request merged**